libs/crypto/hash.hpp

```diff
@@ -19,18 +19,25 @@
 #define IROHA_HASH_H
 
 #include <common/types.hpp>
+#include "model/transaction.hpp"
+#include "model/block.hpp"
+#include "model/query.hpp"
 
 namespace iroha {
 
   void sha3_256(unsigned char *output, unsigned char *input,
                       size_t in_size);
-
   void sha3_512(unsigned char *output, unsigned char *input,
                       size_t in_size);
 
   hash256_t sha3_256(const uint8_t *input, size_t in_size);
-
+  hash256_t sha3_256(const std::string &msg);
   hash512_t sha3_512(const uint8_t *input, size_t in_size);
+  hash512_t sha3_512(const std::string &msg);
+
+  hash256_t hash(const model::Transaction& tx);
```

Comment:
Probably it is better to move this to `irohad/model` since it uses `iroha::model`.

Reply:
I did an attempt and after some thinking decided to leave these functions here. There is no good place for them in existing files so I tried to create a new one. I named it "hash" - looks reasonable for a file with only 3 hashing functions. But now we have 2 files with the same name in different places and their functionality is also about the same so I reverted all changed back.